### DISCUSSION TITLE
insufficient funds for gas * price + value: balance 153177925817010210, tx cost 159824131084000000, overshot 6646205266989790

### DISCUSSION THREAD

**Discussion by owolabi-victor** [Cyfrin/foundry-full-course-cu#2577] [General]
const HDWalletProvider = require("@truffle/hdwallet-provider");
const { Web3 } = require("web3");
const { interface, bytecode } = require("./compile");

const provider = new HDWalletProvider(
  "", // mnemonic
  "" // Infura URL for Sepolia network
);
const web3 = new Web3(provider);

const deploy = async () => {
  try {
    const accounts = await web3.eth.getAccounts();
    console.log("Attempting to deploy from account", accounts[0]);

    // Adjust gas price and limit here
    const gasPrice = web3.utils.toWei('30', 'gwei'); // Set gas price to 2 Gwei
    const gasLimit = "600000"; // Set gas limit to 600,000

    // Deploy the contract with the updated gas price and limit
    const result = await new web3.eth.Contract(JSON.parse(interface))
      .deploy({ data: bytecode, arguments: ["Hi there!"] })
      .send({ gas: gasLimit, gasPrice: gasPrice, from: accounts[0] });

    console.log("Contract deployed to", result.options.address);
  } catch (error) {
    console.error("An error occurred during contract deployment:", error);
  } finally {
    provider.engine.stop();
  }
};

deploy();

//this is the code and its still not deploying after adding a lot of sepolia ether

**Comment by EngrPips** [comment]
Hello @owolabi-victor, Did you figure this out already?
